postgres/parser/parser/sql.y

```diff
@@ -12631,6 +12638,304 @@ a_expr:
 // The UNIQUE predicate is a standard SQL feature but not yet implemented
 // in PostgreSQL (as of 10.5).
 | UNIQUE '(' error { return unimplemented(sqllex, "UNIQUE predicate") }
+// Below here we special-case the OPERATOR(...) syntax only for operators in the pg_catalog schema. 
+// This is to support particular psql commands that require it.
+| a_expr OPERATOR '(' schema_name '.' '+' ')' a_expr
+  {
+    if $4 != "pg_catalog" {
+      sqllex.Error(fmt.Sprintf("schema %q not allowed in OPERATOR syntax", $4))
+      return 1
+    }
```

Comment:
Does this syntax not work for other schemas? It must parse for them since this is a logic check rather than a being enforced within the rule itself. If the `OPERATOR` is supported for other schemas in Postgres, then I'd rather we put the restriction in the `ast` package, rather than being here.

Reply:
It does work for other schemas. I moved the restriction into the ast conversion, good call.